tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -351,6 +352,82 @@ def test_5_inc_snapshots(network_config,
 
     test_matrix.run_test(_test_seq_snapshots)
 
+DEFAULT_HOST_IP = "192.168.0.1"
+DEFAULT_GUEST_IP = "192.168.0.2"
+DEFAULT_TAP_NAME = "tap0"
+DEFAULT_DEV_NAME = "eth0"
+DEFAULT_NETMASK = 30
+
+def test_load_snapshot_failure_handling(bin_cloner_path):
+    """Test scenario:
+    1.  Create a new snapshot from a Firecracker process.
+    2.  Corrupt the snapshot's memory file.
+    3.  Try to build a VM out of the corrupted snapshot.
+    4.  Verify that an error was shown and the process is dead. """
+
+    logger = logging.getLogger("snapshot_load_failure")
+
+    # Use a predefined vm instance.
+    vm_instance = VMNano.spawn(bin_cloner_path)
+    vm = vm_instance.vm
+    root_disk = vm_instance.disks[0]
+    ssh_key = vm_instance.ssh_key
+
+    vm.start()
+    ssh_connection = net_tools.SSHConnection(vm.ssh_config)
+
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    logger.info("Create %s.", SnapshotType.FULL)
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(vm)
+
+    disks = [root_disk.local_path()]
+    # Create base snapshot.
+    snapshot = snapshot_builder.create(disks,
+                                       ssh_key,
+                                       SnapshotType.FULL)
+
+    vm.kill()
+
+    # Corrupt the memory file of the snapshot
+    logger.info("Corrupt the memory file %s", snapshot.mem)
+
+    mem_fd = open(snapshot.mem, "wb")
+    mem_fd.write(bytearray([0xde, 0xad, 0xbe, 0xef, 0x00]))
+    mem_fd.close()
+
+    # Create a new VM instance
+    vm = MicrovmBuilder(bin_cloner_path).get_basevm()
+    vm.spawn(log_level='Info')
+
+    # Hardlink all the snapshot files into the microvm jail.
+    jailed_mem = vm.create_jailed_resource(snapshot.mem)
+    jailed_vmstate = vm.create_jailed_resource(snapshot.vmstate)
+    assert len(snapshot.disks) > 0, "Snapshot requires at least one disk."
+    vm.ssh_config['ssh_key_path'] = snapshot.ssh_key
+
+    vm.create_tap_and_ssh_config(host_ip=DEFAULT_HOST_IP,
+                                    guest_ip=DEFAULT_GUEST_IP,
+                                    netmask_len=DEFAULT_NETMASK,
+                                    tapname=DEFAULT_TAP_NAME)
+
+    response = vm.snapshot_load.put(mem_file_path=jailed_mem,
+                                    snapshot_path=jailed_vmstate)
+
+    logger.info("Response status code %d, content: %s.", response.status_code, response.text)
+    assert vm.api_session.is_status_bad_request(response.status_code)
+
+    logger.info("The Firecracker PID is %d.", vm.jailer_clone_pid)
+    try:
+        os.kill(vm.jailer_clone_pid, 0)
```

Comment:
This is brittle, the pid could be reused, chance is small for that to happen, but I would avoid it if possible.

`os.kill()` can also inadvertently kill other processes that might reuse the pid.

Instead of using `os.kill` you could check process age. See this for example https://unix.stackexchange.com/questions/238641/get-process-age-from-command-line

We can get `jailer_pid` process `start_time` using `/proc/<pid>/stat`, before calling the API. Then recheck after the API call, and it should be either:
 - no process with that PID 👍 
 or
 - process with that PID has newer `start_time` 👍 

Anything else: 👎

Reply:
Understood, will do as such

Reply:
Done!